scripts/components/chat.ts

```diff
@@ -5,13 +5,27 @@ import { MemberData } from 'common/online';
 class ChatHandler {
 	membersTyping: string[] = [];
 	typingLabel = $<Label>('#ChatMemberTypingLabel');
+	history = $('#ChatHistory');
 
 	constructor() {
 		$.RegisterEventHandler('OnNewChatEntry', $.GetContextPanel(), (panel) => this.onNewChatEntry(panel));
 
 		$.RegisterForUnhandledEvent('PanoramaComponent_SteamLobby_OnMemberDataUpdated', (memberData: MemberData) =>
 			this.onSteamLobbyMemberDataUpdated(memberData)
 		);
+
+		// Hacks to scroll to bottom when opening chat / joining/leaving a lobby.
+		// Whether or not scrolling works seems to corresponding to IsReadyForDisplay(), but
+		// I can't get RegisterForReadyEvents/ReadyForDisplay to fire at all.
+		// This timing appears consistent in debug so *hopefully* fine for everyone,
+		// but obviously an event-based approach would be preferred in the future.
+		$.RegisterForUnhandledEvent('HudChat_Show', () => {
+			$.Schedule(0.1, () => this.history.ScrollToBottom());
+		});
+
+		$.RegisterForUnhandledEvent('PanoramaComponent_SteamLobby_OnLobbyStateChanged', () =>
```

Comment:
This is going to scroll to the bottom of the chat every time there's a lobby state change, so this will interrupt people scrolling up and reading chat. Is that intended?

Reply:
Pretty sure the only two state changes that can happen here are you joining or leaving, but I'll double check and add explicit checks here if not.